src/main/io/osd.c

```diff
@@ -2787,47 +2806,59 @@ static bool osdDrawSingleElement(uint8_t item)
                 case OSD_UNIT_UK:
                     FALLTHROUGH;
                 case OSD_UNIT_IMPERIAL:
-                    moreThanAh = osdFormatCentiNumber(buff, value * METERS_PER_MILE / 10, 1000, 0, 2, 3);
+                    moreThanAh = osdFormatCentiNumber(buff, value * METERS_PER_MILE / 10, 1000, 0, 2, digits);
                     if (!moreThanAh) {
                         tfp_sprintf(buff, "%s%c%c", buff, SYM_MAH_MI_0, SYM_MAH_MI_1);
                     } else {
                         tfp_sprintf(buff, "%s%c", buff, SYM_AH_MI);
                     }
                     if (!efficiencyValid) {
-                        buff[0] = buff[1] = buff[2] = '-';
-                        buff[3] = SYM_MAH_MI_0;
-                        buff[4] = SYM_MAH_MI_1;
-                        buff[5] = '\0';
+                        if(bf_compat){
+                            buff[0] = buff[1] = buff[2] = buff[3] = '-';    
+                        } else {
+                            buff[0] = buff[1] = buff[2] = '-';
+                        }                        
+                        buff[digits] = SYM_MAH_MI_0;
+                        buff[digits + 1] = SYM_MAH_MI_1;
+                        buff[digits + 2] = '\0';
                     }
                     break;
                 case OSD_UNIT_GA:
-                     moreThanAh = osdFormatCentiNumber(buff, value * METERS_PER_NAUTICALMILE / 10, 1000, 0, 2, 3);
+                     moreThanAh = osdFormatCentiNumber(buff, value * METERS_PER_NAUTICALMILE / 10, 1000, 0, 2, digits);
                     if (!moreThanAh) {
                         tfp_sprintf(buff, "%s%c%c", buff, SYM_MAH_NM_0, SYM_MAH_NM_1);
                     } else {
                         tfp_sprintf(buff, "%s%c", buff, SYM_AH_NM);
                     }
                     if (!efficiencyValid) {
-                        buff[0] = buff[1] = buff[2] = '-';
-                        buff[3] = SYM_MAH_NM_0;
-                        buff[4] = SYM_MAH_NM_1;
-                        buff[5] = '\0';
+                        if(bf_compat){
+                            buff[0] = buff[1] = buff[2] = buff[3] = '-';    
+                        } else {
+                            buff[0] = buff[1] = buff[2] = '-';
+                        }
```

Comment:
I don't think you need this if statement. Just using 
```
if (!efficiencyValid) {
    buff[0] = buff[1] = buff[2] = buff[3] = '-';
    buff[digits] = SYM_MAH_NM_0;
...etc
```
would be fine. As buff[3] will be overwritten by the symbol, if not using BF43COMPAT.

Reply:
There are other if statements in that block doing the same.

Reply:
Oh yes, you are totally right! That indeed will streamline the code. Good catch!

Reply:
Made these changes in [f4b7f57](https://github.com/iNavFlight/inav/pull/8866/commits/f4b7f574d144fdae2198ff1e2a2ac18d83c7e2e6)